text/0023-stats-component.md

```diff
@@ -0,0 +1,345 @@
+  valid - the number of cells in that column that contain valid strings
+  error - the number of cells in that column that contain invalid string
+  missing - the number of cells in that column that do not contain data 
+  minLength - the shortest length of a string in that column
+  maxLength - the longest length of a string in that column
+
+Boolean
+  valid - the number of cells in that column that contain valid boolean values
+  error - the number of cells in that column that contain invalid boolean values
+  missing - the number of cells in that column that do not contain data 
+  trueCount - the number of cells that contain a true value
+  falseCount - the number of cells that contain a false value
+
+The structure of the default stats component depends on the structure of the data itself. If each row of the dataset body is expressed as an array, the stats component will be an array of objects. If each row of the dataset body is expressed as an object, the stats component will be an object of objects. One important thing to note, we must have a structure with a schema that describes the title and the type in order to create stats.
+
+Let's look at two examples of a small dataset body, and the resulting stats object.
+
+```json
+// this is a small dataset body expressed as a 2D array. It is a list of 
+// classes in an elementary school. The columns are the class name (string),
+// the current number of students (number), and whether the class can accept 
+// more students
+[
+  ["1A", 20, false],
+  ["1B", 22, false],
+  ["1C", 17, true],
+  ["1D", 19, true]
+]
+
+// the resulting stats section would look like this:
+[
+  {
+    "title":"class name",
+    "type":"string",
+    "count": 4,
+    "minLength": 2,
+    "maxLength": 2
+  },
+  {
+    "title":"students",
+    "type":"number",
+    "count": 4,
+    "min": 17,
+    "max": 22,
+    "avg": 19.5,    
+  },
+  {
+    "title":"can accept more students",
+    "type":"boolean",
+    "count": 4,
+    "trueCount": 2,
+    "falseCount": 2,
+  }
+]
+```
+Note that the data we expressed as a 2D array, and so the stats component is presented as an array.
+
+Let's look at this same example, but with the data presented as an array of objects:
+
+```json
+[
+  {
+    "class_name":"1A",
+    "students":20, 
+    "can_accept_more":false
+  },
+  {
+    "class_name":"1B",
+    "students":22, 
+    "can_accept_more":false
+  },
+  {
+    "class_name":"1C",
+    "students":27, 
+    "can_accept_more":true
+  },
+  {
+    "class_name":"1D",
+    "students":19, 
+    "can_accept_more_students":true
+  }
+]
+
+// stats:
+{
+  "class_name": {
+    "title":"class name",
+    "type":"string",
+    "count": 4,
+    "minLength": 2,
+    "maxLength": 2
+  },
+  "students": {
+    "title":"students",
+    "type":"number",
+    "count": 4,
+    "min": 17,
+    "max": 22,
+    "avg": 19.5,    
+  },
+  "can_accept_more_students": {
+    "title":"can accept more students",
+    "type":"boolean",
+    "count": 4,
```

Comment:
just to point out, I kept 'count' here, rather then 'valid', 'error', and 'missing'

waiting for feedback on that before I standardize!